src/server/editorServices.ts

```diff
@@ -1002,7 +1002,13 @@ namespace ts.server {
                 directory,
                 fileOrDirectory => {
                     const fileOrDirectoryPath = this.toPath(fileOrDirectory);
-                    project.getCachedDirectoryStructureHost().addOrDeleteFileOrDirectory(fileOrDirectory, fileOrDirectoryPath);
+                    const fileSystemResult = project.getCachedDirectoryStructureHost().addOrDeleteFileOrDirectory(fileOrDirectory, fileOrDirectoryPath);
+
+                    // don't trigger callback on open, existing files
+                    if (project.fileIsOpen(fileOrDirectoryPath) && fileSystemResult && fileSystemResult.fileExists) {

```

Comment:
Why the extra checks?

Reply:
Without the extra checks, the test on line 17 of `syntaxOperations.ts` fails. It seems to be explicitly testing that when an open file is deleted but not closed, `updateGraph` should run. This seemed reasonable to me, so I chose to handle the case instead of changing the test. What do you think?

Reply:
After discussing in person, we changed the test instead of the behavior.